checkmate/checker/pipeline/file.py

```diff
@@ -0,0 +1,67 @@
+"""Stages which read files."""
+
+import csv
+from logging import getLogger
+from zipfile import BadZipFile, ZipFile
+
+from checkmate.checker.pipeline.core import Stage
+from checkmate.exceptions import StageException
+
+LOG = getLogger(__name__)
+
+
+class ReadTextFile(Stage):
+    """A stage which reads a file and outputs a generator of lines."""
+
+    def __init__(self, name=None):
+        self._name = name
+
+    def __call__(self, working_dir, source=None):
+        if source is None:
+            source = self._name
+
+        self.check_file(source)
+
+        with open(source) as handle:
+            yield from handle
+
+
+class ReadCSVFile(ReadTextFile):
+    """A stage which reads a CSV file and outputs a generator of rows."""
+
+    def __call__(self, working_dir, source=None):
+        lines = super().__call__(working_dir, source)
+        cleaned_lines = (line for line in lines if not line.startswith("#"))
```

Comment:
The CSV files from URLHaus have non-standard lines with comments in. I'd guess that's common.